File: app/src/main/java/cn/com/woong/readhub/ui/WebActivity.java
package cn.com.woong.readhub.ui;

import android.content.Context;
import android.content.Intent;
import android.widget.FrameLayout;

import com.blankj.utilcode.util.BarUtils;
import com.just.agentweb.AgentWeb;

import butterknife.BindView;
import cn.com.woong.readhub.R;
import cn.com.woong.readhub.base.BaseActivity;

/**
 * Created by wong on 2018/4/20.
 */

public class WebActivity extends BaseActivity {
    @BindView(R.id.fl_web)
    FrameLayout flWeb;

    private static final String WEB_URL = "web_url";
    private String mWebUrl;

    AgentWeb mAgentWeb;

    public static void startWebActivity(Context context, String url) {
        Intent intent = new Intent(context, WebActivity.class);
        intent.putExtra(WEB_URL, url);
        context.startActivity(intent);
    }

    @Override
    protected int getLayoutId() {
        return R.layout.activity_web;
    }

    @Override
    protected void initView() {
        BarUtils.setStatusBarVisibility(this, false);
    }

    @Override
    protected void initData() {
        mWebUrl = getIntent().getStringExtra(WEB_URL);

        mAgentWeb = AgentWeb.with(this)
                .setAgentWebParent(flWeb, new FrameLayout.LayoutParams(-1, -1))
                .useDefaultIndicator()
                .createAgentWeb()
                .ready()
                .go(mWebUrl);
    }
}

File: app/src/main/java/cn/com/woong/readhub/ui/WebActivity.kt
package cn.com.woong.readhub.ui

import android.content.Context
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import cn.com.woong.readhub.R
import com.blankj.utilcode.util.BarUtils
import com.just.agentweb.AgentWeb
import android.widget.FrameLayout
import kotlinx.android.synthetic.main.activity_web.*
import android.content.Intent

class WebActivity : AppCompatActivity() {
    private val WEB_URL = "web_url"
    private var mWebUrl: String? = null

    var mAgentWeb: AgentWeb? = null

    fun startWebActivity(context: Context, url: String) {
        val intent = Intent(context, WebActivity::class.java)
        intent.putExtra(WEB_URL, url)
        context.startActivity(intent)
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_web)

        BarUtils.setStatusBarVisibility(this, false)

        mWebUrl = intent.getStringExtra(WEB_URL)

        mAgentWeb = AgentWeb.with(this)
                .setAgentWebParent(fl_web, FrameLayout.LayoutParams(-1, -1))
                .useDefaultIndicator()
                .createAgentWeb()
                .ready()
                .go(mWebUrl)
    }
}